backends/lean/Base/Progress/Progress.lean

```diff
@@ -343,17 +335,27 @@ def evalProgress (args : TSyntax `Progress.progressArgs) : TacticM Stats := do
     if withArg.size > 0 then
       let id := withArg.get! 1
       trace[Progress] "With arg: {id}"
-      -- Attempt to lookup a local declaration
-      match (← getLCtx).findFromUserName? id.getId with
-      | some decl => do
-        trace[Progress] "With arg: local decl"
-        pure (some (.Local decl))
-      | none => do
-        -- Not a local declaration: should be a theorem
-        trace[Progress] "With arg: theorem"
-        addCompletionInfo <| CompletionInfo.id id id.getId (danglingDot := false) {} none
-        let some (.const name _) ← Term.resolveId? id | throwError m!"Could not find theorem: {id}"
-        pure (some (.Theorem name))
+
+      -- Either, it's a local declaration or a theorem, either it's a term
+      -- for which we need to find a declaration
+
+      if id.isIdent then
```

Comment:
I don't think we really need this case disjunction. Shouldn't it be performed by Lean's elaboration (i.e., isn't the `else` branch enough)?

Reply:
Hm, if I do that, something like:
```
set_option trace.Progress true in
  example {x: U32}
    (f : Result U32 → Result Unit)
    (h : ∀ (x: U32), ∀ (y: U32), f (x + y) = .ok ()):
      f (.ok x) = ok () := by
      have : ∀ x, f x = .ok () := by sorry
      progress with this
```

won't work because `this` elaborated := `sorryAx ?m.something true` and the metavariable will create an obscure failure of progress here.

Reply:
(tbh, the current code does not work neither, it fails by not finding `this` in the context)

Reply:
Ok, this is good to know (and worth commenting in the code ;) )

Reply:
Wouldn't this be a blocker though for the PR? I'm trying to make this work but I'm just diving into how the whole thing is supposed to work

Reply:
What I meant is that the fact that we need the case disjunction is good to know and should be documented. But of course, we need to make the elaboration work, generally speaking.